packages/core/utils/src/validate/index.ts

```diff
@@ -26,11 +26,18 @@ export interface ValidateFunc {
   (data: unknown, schema: Model, options?: Options): Promise<void>;
 }
 
+interface APIOptions {
+  validators?: Validators;
+  getModel: (model: string) => Model;
+}
+
 export interface Validators {
   input?: Validator[];
 }
 
-const createAPIValidators = (opts?: { validators: Validators }) => {
+const createAPIValidators = (opts: APIOptions) => {
+  const { getModel } = opts || {};
```

Comment:
If opts can be falsy, doesn't that mean it's optional? Why not using `opts: APIOptions = {} `?

Reply:
the getModel is required in the typings so doing this would fail. but at runtime for non ts caller you could send nothing. this is the kinda of place where ts actually makes your work harder as it assumes this code will be called with the right type but it might not

Reply:
> with the right type but it might not

This is also where we should draw the line between public and internal APIs IMO. Public APIs inputs should always be validated, whereas I would argue internal APIs (except critical ones of course) should rely on TypeScript "correctness".

Reply:
I agree, it's too public for now